### PR TITLE
Added SLDS Slider

### LINKED ISSUE #1314: Add Slider Component
Initial request for a slider component. Support the following props:
- id
- name
- label
- value
- min
- max
- steps
- disabled
- size
- vertical (default false)
- className
- errorText
- aria-describedby
- assistiveText
- onChange
- onInput

### PR DESCRIPTION
**Pull request by DanFerro**
This adds the [SLDS Slider component](https://www.lightningdesignsystem.com/components/slider/) to the component library.

Fixes #1314

@interactivellama 

---

### Pull Request Review checklist (do not remove)

* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [x] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at snapshot strings.
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [x] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [x] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

File: components/slider/index.jsx [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+	/**
+	 * Indicates the granularity that is expected by limiting the allowed values.
+	 */
+	step: PropTypes.number,
+	/**
+	 * Disables the Slider and prevents clicking it.
+	 */
+	disabled: PropTypes.bool,
+	/**
+	 * Size of the slider.
```
Comment (interactivellama) [inline components/slider/index.jsx]: We should mention this is only horizontal size.

File: components/slider/index.jsx [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	getId () {
+		return this.props.id || this.generatedId;
+	}
+
+	getErrorId () {
+		return this.props['aria-describedby'] || this.generatedErrorId;
+	}
+
+	handleChange = (event) => {
+		this.setState(({ value: event.target.value }));
```
[the remark this answers is not in the document]
Comment (DanFerro) [reply components/slider/index.jsx]: @interactivellama just to clarify, the user should be the one in charge of making sure that span.slds-slider__value is up to date with the current slider value(as per implementation note https://www.lightningdesignsystem.com/components/slider/)?

[the remark this answers is not in the document]
Comment (interactivellama) [reply components/slider/index.jsx]: To be a controlled component, the user should be triggering an `onChange` event that passes the value up to a state container (or Redux in an app) that then passes it down as the `props.value`. You can see this in other examples, for instance, a controlled `<Input/>` textbox. The keyPress fires of the change event that then changes the value and loops it back in. This allows the component to be a functional stateless components.

**Comment by DanFerro** [comment]
@interactivellama should be good. Let me know if I missed anything.

File: components/slider/index.jsx [hunk 3]
```diff
@@ -24,81 +24,91 @@ import { SLIDER } from '../../utilities/constants';
...
 	 */
-	className: PropTypes.oneOfType([
-		PropTypes.array,
-		PropTypes.object,
-		PropTypes.string,
-	]),
+	name: PropTypes.string,
 	/**
-	 * Message to display when the Slider is in an error state. When this is present, also visually highlights the component as in error.
+	 * This event fires whenever the user has modified the data of the control.
```
Comment (interactivellama) [inline components/slider/index.jsx]:
```
* This event fires whenever the user has modified the data of the control. `event, { value: [string] }` is passed in.
```

See https://github.com/salesforce/design-system-react/blob/master/components/combobox/combobox.jsx#L427

Comment (interactivellama) [reply components/slider/index.jsx]: Oh, sorry. That's exactly what you did. The [tests](https://github.com/salesforce/design-system-react/pull/1317/commits/9529d9afc2d60002321fabf02f910b5ab7eae71b#diff-180f22e91382498f7fbe067108b3a63eR187) though seem to not get an object, but a straight number. The test probably just needs updating.

Comment (DanFerro) [reply components/slider/index.jsx]: @interactivellama can you clarify which prop you're referring to?

Comment (DanFerro) [reply components/slider/index.jsx]: Updated to follow the combobox implementation, passing an object with key of value.

**Pull request merged**